docs/intro/overrides.rst

```diff
@@ -58,84 +85,223 @@ Let's take a look at how the following code is structured:
+          Item Class is returned by the ``DualExampleProductPage``'s ``to_item()``
+          method instead.
+
+This has the same concept as with the Page Object Overrides but the context changes
+from using a Page Object into Item Classes instead (returned by the ``to_item()``
+method).
+
+The upside here is that you're able to directly access the item extracted by the
+Page Object for the given site. There's no need to directly call the ``to_item()``
+method to receive it.
+
+However, one downside to this approach is that you lose access to the actual Page
+Object. Aside from the item extracted by the Page Object, there might be some
+other convenience methods or other data from it that you want to access.
+
+
+.. _combination:
+
+Combination
+~~~~~~~~~~~
+
+Of course, you can use the combination of both which enables you to specify in
+either contexts of Page Objects and Item Classes.
+
+.. code-block:: python
+
+    from web_poet import handle_urls, WebPage
+
+
+    class GenericProductPage(WebPage[Product]):
+        def to_item(self) -> Product:
+            return Product(product_title=self.css("title::text").get())
+
+
+    @handle_urls("some.example", instead_of=GenericProductPage, to_return=Product)
+    class ExampleProductPage(WebPage):
+        ...  # more specific parsing
+
+
+    @handle_urls("another.example", instead_of=GenericProductPage, exclude="/digital-goods/")
+    class AnotherExampleProductPage(WebPage[Product]):
+        ...  # more specific parsing
+
+
+    @handle_urls(["dual.example/shop/?product=*", "uk.dual.example/store/?pid=*"], instead_of=GenericProductPage)
+    class DualExampleProductPage(WebPage[SimilarProduct]):
+        ...  # more specific parsing
+
+To recap some previously mentioned behaviors:
+
+    - In the ``@handle_urls`` decorator for ``ExampleProductPage``, we needed to
+      pass the ``to_return`` parameter since the Page Object didn't specify the
+      Item Class that it returns (i.e. ``class ExampleProductPage(WebPage[Product])``).
+      If it did, we could omit the ``to_return`` parameter entirely since it's
+      able to automatically derive the Item Class.
+    - The recommended way is to avoid using the ``to_return`` parameter in
+      ``@handle_urls`` entirely (similar to how ``AnotherExampleProductPage``
+      and ``DualExampleProductPage`` does it).
+
+.. warning::
+
+    Only use the ``to_return`` parameter in the ``@handle_urls`` decorator for
+    more advanced use cases since it presents the risk of having the Item Class
+    returned by the Page Object diverge with the actual ``to_return`` value in
+    the :class:`~.ApplyRule`. For example:
+
+    .. code-block:: python
+
+        @handle_urls("some.example", to_return=DifferentProduct)
+        class ProductPage(WebPage[Product])
+            ... # more specific parsing
+
+    The Page Object returns a ``Product`` item but the created :class:`~.ApplyRule`
+    has ``to_return=DifferentProduct``.
```

Comment:
> I don’t like that we mention “more advanced use cases” without specific examples of such use cases. I would either not mention the existence of such use cases at all, or add a later section that covers one such case, and link to that here.

You're right. Although adding such advanced use cases might encourage users to go that route. On the other hand, omitting the availability of the `to_return` option might not provide users with a clue that it's available for `@handle_urls`. Although I think the API refs showing it available as one of the parameters is already enough.

With this, I'm inclined to remove the mention of `to_return` in the docs. It could also simplify this tutorial a lot more. This would also promote the much preferred way of declaring them via:
```python
@handle_urls(URL)
class ProductPage(WebPage[Product]): ...
```

@kmike do you have some specific cases in mind that you think we should mention? If none, we can remove `to_return` from the docs.

Reply:
> @kmike do you have some specific cases in mind that you think we should mention? If none, we can remove to_return from the docs.

I think that's fine to "undocument" to_return parameter of `handle_urls`.

Reply:
Removed `to_return` usage in `@handle_urls()` https://github.com/scrapinghub/web-poet/pull/84/commits/076e7bb5ae74b700789ea8a2e74882d66253747c